internal/cluster/transition.go

```diff
@@ -142,33 +145,146 @@ func (th *transitionHandler) PostUpdateInstallationProgress(sw stateswitch.State
 // Complete installation
 ////////////////////////////////////////////////////////////////////////////
 
-type TransitionArgsCompleteInstallation struct {
-	ctx       context.Context
-	isSuccess bool
-	reason    string
-}
-
 func (th *transitionHandler) PostCompleteInstallation(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) error {
 	sCluster, ok := sw.(*stateCluster)
 	if !ok {
 		return errors.New("PostCompleteInstallation incompatible type of StateSwitch")
 	}
-	params, ok := args.(*TransitionArgsCompleteInstallation)
+	params, ok := args.(*TransitionArgsRefreshCluster)
 	if !ok {
 		return errors.New("PostCompleteInstallation invalid argument")
 	}
 
-	return th.updateTransitionCluster(logutil.FromContext(params.ctx, th.log), th.db, sCluster, params.reason)
+	status, reason := th.getClusterMonitoringOperatorsStatus(sCluster.cluster)
+
+	return th.updateCompletion(params.ctx, params.ocmClient, params.metricApi, params.eventHandler, sCluster, status == models.OperatorStatusAvailable, reason)
 }
 
-func (th *transitionHandler) isSuccess(stateSwitch stateswitch.StateSwitch, args stateswitch.TransitionArgs) (b bool, err error) {
-	params, _ := args.(*TransitionArgsCompleteInstallation)
-	return params.isSuccess, nil
+func (th *transitionHandler) updateCompletion(ctx context.Context, ocmClient *ocm.Client, metricAPI metrics.API, eventsHandler events.Handler,
```

Comment:
why it should be part of transitionHandler ? what it uses from it?

Reply:
It's not mandatory. Just followed all the other private functions signatures